### ISSUE TITLE
endSession doesn't work after transfer

### ISSUE THREAD

**Issue by Loyc59** [MyPureCloud/genesys-cloud-webrtc-sdk#492]
Hello,

We are integrating the sdk in our CRM and all our cheklist is green except one use case.

The call to the **endSession** method (with sessionId or conversationId) does nothing after a transfer.
After an incoming call (no transfer) it is working as expected.

There is no warning, no error.

Here are the logs when it is working (incoming call no transfer) :
![image](https://user-images.githubusercontent.com/3726236/112839860-f257b980-909e-11eb-9cf6-15ba22a791c0.png)

Here are the logs when it is not working (transfer) :
![image](https://user-images.githubusercontent.com/3726236/112840262-6abe7a80-909f-11eb-908f-62ac1064b689.png)

This problem occurs at least on v4.1.2 and on v5.0.5.

Best

**Comment by jensengar** [comment]
I have a few questions for you. 

1. When you say `The call to the endSession method (with sessionId or conversationId) does nothing after a transfer` does that mean the call is still completely connected and you can still talk to the other party?
2. Are you using persistent connection?

In the meantime, I will see if we can reproduce this.

**Comment by Loyc59** [comment]
1. Yes exactly. The only way to end the call is to wait for the other party to end it on its phone.
2. What do you mean ?

I have made some other tests and I have the same behavior with **setAudioMute** method.

Here the normal behavior with an incoming call (no transfer) > muted has the expected effect on the conversation : 
![image](https://user-images.githubusercontent.com/3726236/112968952-1671d400-914d-11eb-947c-e06f3af91d7a.png)

Here the problem when the call has been transfered > muted has no effect on the conversation : 
![image](https://user-images.githubusercontent.com/3726236/112969104-3b664700-914d-11eb-9c29-a09344f2298d.png)

The muted flag stays false.

Maybe it tries to modify the previous agent ?

**Comment by jensengar** [comment]
interesting... That's possible. Let me have another go at reproducing this tomorrow. Thanks for the info, this is very helpful.

**Comment by jensengar** [comment]
I just tried to reproduce this using an ivr and queues and separate users just like you have here and I wasn't able to see what you are seeing. I'd like to look at the logs and see if there's anything there, can you tell me what region this is in?

**Comment by Loyc59** [comment]
France and we use the german env : 

`sdk = new $window.GenesysCloudWebrtcSdk({
    accessToken: accessToken,
    environment: 'mypurecloud.de'
});`

**Comment by jensengar** [comment]
I didn't see any problems in the logs, @Loyc59 can you look at the network tab and verify you are sending an api request to end the call? It will look something like this:

```
PATCH /conversations/calls/${conversationId}/participants/${participantId}
```
body will look like this:
```
{ state: 'disconnected' }
```

**Comment by Loyc59** [comment]
Yes I see a PATCH with a success code but I am still in conversation, it has no effect.

Thu, 08 Apr 2021 13:29:50 GMT

PATCH https://api.mypurecloud.de/api/v2/conversations/calls/da6fe467-3d18-4042-ab41-b6e63b773cb1/participants/079f4d35-5cbd-42ea-9953-86da179476fa
Body = {state: "disconnected"}

Response code : 202 Accepted

I even did it 6 times in a row but still 202 and no effect.

Thanks for your perseverance on this subject ;)

**Comment by jensengar** [comment]
ok I see the problem. When we try and end the call, we are trying to find the participant like this:
```
const participant = participants.find((p) => p.userId === this.sdk._personDetails.id);
```
but that doesn't work in this case because both the participants are from the same user so it's finding the first user. I'll get a pr up with better logic for identifying the participant.

**Comment by jensengar** [comment]
https://github.com/MyPureCloud/genesys-cloud-webrtc-sdk/pull/507

**Comment by jensengar** [comment]
this is fixed in newly released version 5.0.9.

**Comment by Loyc59** [comment]
> this is fixed in newly released version 5.0.9.

Thanks @jensengar it works now.
